### ISSUE TITLE
NIRISSBackwardGrismDispersion crashes with old calib files

### ISSUE THREAD

**Issue by Russell-Ryan** [spacetelescope/jwst#7452]
The file transforms/models.py at lines 927 and 928, which are:
```
        dx = xmodel[0](x, y) + t * xmodel[1](x, y) + t**2 * xmodel[2](x, y)
        dy = ymodel[0](x, y) + t * ymodel[1](x, y) + t**2 * ymodel[2](x, y)
```
are hard-coded polynomials in the grism dispersion parameter t.  The new calib files have second order parameters xmodel and ymodel, but the earlier versions only have linear models.  Therefore, the new code with the old files cause a crash, which makes comparison between the different reference files very difficult.  These lines could be generalized and avoid the hard-coded assumption of the parameter's order, such as:

```
        dx=sum(m(x,y)*t**i for i,m in enumerate(xmodel))
        dy=sum(m(x,y)*t**i for i,m in enumerate(ymodel))
``` 
And now the same piece of code is compatible with different vintage reference files.  

Is this possible?

**Comment by hbushouse** [comment]
Anything is possible, but not necessarily probable!  ;-)

But seriously, in general we don't support ref files that are in an old format with newer versions of the cal code. The version of the software that's in use is expected to be matched to the CRDS context of the appropriate/matching reference files. In other words, if you want to use obsolete ref files, you need to use an old version of the code that matches them (and vice versa).

However, we are in the process right now of updating the use of WFSS transforms in the cal code to accommodate the most recent form of the calibrations received from the NIRCam team, based on the latest in-flight calibrations they've derived. So it might be possible to work in a more generic solution like you describe above while we're doing those updates. Whether it's possible will depend on the exact form of the latest transforms.

**Comment by hbushouse** [comment]
Adding this to https://jira.stsci.edu/browse/JP-2113, which is tracking the work to update the NIRCam WFSS transforms in the Cal code.

**Comment by hbushouse** [comment]
> But this is about NIRISS not, NIRCam. I don't think there is any serious change, it's only not explicitly evaluating a polynomial, but using pythonic tools.

Yes, sorry about that. Noticed that myself a few minutes after writing my response. But I think it could still be worked as part of the other updates to transforms that we're in the process of doing for NIRCam, even though this one is specific to NIRISS.

**Comment by melanieclarke** [comment]
Closing as a non-issue: using old reference files also requires using an old version of the pipeline code.
